testsuite/features/support/env.rb

```diff
@@ -127,31 +127,73 @@ def capybara_register_driver
   $feature_scope = scenario.location.file.split(%r{(\.feature|/)})[-2]
 end
 
-# embed a screenshot after each failed scenario
+# Embed a screenshot after each failed scenario
 After do |scenario|
   current_epoch = Time.new.to_i
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
+  log "Current url is #{current_url}"
   if scenario.failed?
     begin
-      Dir.mkdir('screenshots') unless File.directory?('screenshots')
-      path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
-      # only click on Details when we have errors during bootstrapping and more Details available
-      click_button('Details') if has_content?('Bootstrap Minions') && has_content?('Details')
-      # a Timeout::Error may be raised while a page is still (re)loading
-      find('#page-body', wait: 3) if scenario.exception.is_a?(Timeout::Error)
+      if web_session_is_active?
+        handle_screenshot_and_relog(scenario, current_epoch)
+      else
+        warn 'Page is not visible; unable to take a screenshot.'
+      end
+    ensure
+      print_server_logs
+    end
+  end
+  page.instance_variable_set(:@touched, false)
+end
+
+# Test is web session is open
+def web_session_is_active?
+  # When no Web Session is open, current_url is equal to data:,
+  return false if current_url.empty? || current_url == 'data:,'
```

Comment:
We can easily check the current url to verify if a session is active.
The only issue is current_url is equal to `data:,` when no websession is active.

Reply:
Could `Capybara::Session.instance_created?` fit better here to identify if a session is active?